packages/muelu/test/scaling/Driver.cpp

```diff
@@ -621,24 +623,19 @@ int main_(Teuchos::CommandLineProcessor &clp, Xpetra::UnderlyingLib& lib, int ar
           }
 
           // Belos parameter list
-          Teuchos::ParameterList belosList;
-          belosList.set("Maximum Iterations",    maxIts); // Maximum number of iterations allowed
-          belosList.set("Convergence Tolerance", tol);    // Relative convergence tolerance requested
-          belosList.set("Verbosity",             Belos::Errors + Belos::Warnings + Belos::StatusTestDetails);
-          belosList.set("Output Frequency",      1);
-          belosList.set("Output Style",          Belos::Brief);
+          RCP<Teuchos::ParameterList> belosList = Teuchos::parameterList();
+          belosList->set("Maximum Iterations",    maxIts); // Maximum number of iterations allowed
+          belosList->set("Convergence Tolerance", tol);    // Relative convergence tolerance requested
+          belosList->set("Verbosity",             Belos::Errors + Belos::Warnings + Belos::StatusTestDetails);
+          belosList->set("Output Frequency",      1);
+          belosList->set("Output Style",          Belos::Brief);
           if (!scaleResidualHist)
-            belosList.set("Implicit Residual Scaling", "None");
+            belosList->set("Implicit Residual Scaling", "None");
 
           // Create an iterative solver manager
-          RCP< Belos::SolverManager<SC, MV, OP> > solver;
-          if (solveType == "cg") {
-            solver = rcp(new Belos::PseudoBlockCGSolMgr   <SC, MV, OP>(belosProblem, rcp(&belosList, false)));
-          } else if (solveType == "gmres") {
-            solver = rcp(new Belos::BlockGmresSolMgr<SC, MV, OP>(belosProblem, rcp(&belosList, false)));
-          } else if (solveType == "bicgstab") {
-            solver = rcp(new Belos::BiCGStabSolMgr<SC, MV, OP>(belosProblem, rcp(&belosList, false)));
-          }
+          Belos::SolverFactory<SC, MV, OP> solverFactory;
```

Comment:
Note that this requires instantiating all of Belos' solvers for Xpetra objects.  It also excludes the Tpetra-specific solvers that Ichi and I have been adding, like pipelined CG and s-step GMRES.  In a future PR, it would be good for MueLu to wrap Belos::SolverFactory, so that it only ever gets instantiated for Epetra and Tpetra.